lib/ssoe/service.rb

```diff
@@ -30,27 +25,28 @@ def get_traits(credential_method:, credential_id:, user:, address:)
         parse_response(raw_response.body)
       end
     rescue Common::Client::Errors::ClientError => e
-      error_response(e, :client, e.status)
-    rescue *CONNECTION_ERRORS => e
-      return parse_response(e.response.body) if e.respond_to?(:response) && e.response&.body
-
-      error_response(e, :connection, 502)
+      error_response(e, SSOe::Errors::RequestError, :client)
+    rescue Faraday::ConnectionFailed => e
+      error_response(e, SSOe::Errors::ConnectionError, :connection)
+    rescue Faraday::TimeoutError => e
+      error_response(e, SSOe::Errors::TimeoutError, :timeout)
+    rescue Common::Exceptions::GatewayTimeout => e
+      error_response(e, SSOe::Errors::TimeoutError, :gateway_timeout)
+    rescue Breakers::OutageException => e
+      error_response(e, SSOe::Errors::ConnectionError, :outage)
     rescue => e
-      error_response(e, :unknown, 500)
+      error_response(e, SSOe::Errors::UnknownError, :unknown)
     end
```

Comment:
You can just rescue and re-raise the errors classes here directly instead of calling a separate method. We don't really need to log a message in this class since it will get logged downstream when it's rescued (in the job)